jsk_baxter_robot/baxtereus/test/test-baxter.l

```diff
@@ -69,6 +87,108 @@
     (assert (eps-v= (send robot :angle-vector) av 5) "unable to revert after ik failed")
     ))
 
+(deftest test-robot-ik-bin-cube
+  (let (robot)
+    (setq robot (instance baxter-robot :init))
+    (send (send robot :rarm :end-coords)
+          :newcoords (make-coords :pos #f(50 0 310)
+                                  :rpy (float-vector 0 -pi/2 0)))
+    (send (send robot :larm :end-coords)
+          :newcoords (make-coords :pos #f(50 0 310)
+                                  :rpy (float-vector 0 -pi/2 0)))
+    (send robot :locate #f(0 0 950) :world)
+    (objects (append (list robot) *pod-bin-cube*))
+    (setq step-dist 100)
+    (dolist (cube-data (mapcar #'list *pod-bin-ik-limit* *pod-bin-cube*))
+      (setq limit (car cube-data))
+      (setq cube (cadr cube-data))
+      (setq x (elt (send (send cube :worldcoords) :pos) 0))
+      (setq y (elt (send (send cube :worldcoords) :pos) 1))
+      (setq z (elt (send (send cube :worldcoords) :pos) 2))
+      (setq dx (x-of-cube cube))
+      (setq dy (y-of-cube cube))
+      (setq dz (z-of-cube cube))
+      (do ((tmp-x (+ (- x (/ dx 2)) (/ step-dist 2)) (+ tmp-x step-dist)))
+        ((> tmp-x (+ x (/ dx 2))) t)
+        (do ((tmp-y (+ (- y (/ dy 2)) (/ step-dist 2)) (+ tmp-y step-dist)))
+          ((> tmp-y (+ y (/ dy 2))) t)
+          (do ((tmp-z (+ (- z (/ dz 2)) (/ step-dist 2)) (+ tmp-z step-dist)))
+            ((> tmp-z (+ z (/ dz 2))) t)
+            (setq pos (float-vector tmp-x tmp-y tmp-z))
+            (assert
+              (or
+                (send robot :rarm :inverse-kinematics
+                  (make-cascoords :pos (v+ pos #f(-90 0 0)))
+                  :rotation-axis t
+                  :debug-view :no-message)
+                (send robot :larm :inverse-kinematics
+                  (make-cascoords :pos (v+ pos #f(-90 0 0)))
+                  :rotation-axis t
+                  :debug-view :no-message)
+                (> tmp-x limit))
```

Comment:
why you only check the lower limit of the position? you do not need something like `(< tmp-x limit2)` ??